### PR TITLE
Change Create Flyer route to use form data for image upload

### PR DESCRIPTION
**Pull request by zachseidner1**
## Overview

Added Express route using form-data to post Flyers. Wasn't sure where to put it so I just put it in `app.ts` so I had easy access to the app. This could definitely be the wrong place though and I want feedback on this. Also I know the code may not be the cleanest, and obviously there's still stuff leftover from the previous create flyer endpoint that is now going unused, so we can either clean everything up later or you all can advise me now in this PR.

## Changes Made

- Added post endpoint for `/flyers/` that takes an image file as form data and puts it on the upload service.

## Test Coverage

I used Postman to test this feature and it worked as expected. I tried removing required fields, tried sending and not sending the file, everything works as expected to my knowledge.

## Next Steps 
- Organize this route in the right place in the code base.
- Clean up any remaining code related to creating flyers (remove it if it's being unused)

## Screenshots (delete if not applicable)
image \
Not sure why the image removal is failing but that's not really relevant for what we are trying to do, I think that was failing before I made changes?
image

### CODE REVIEW

File: src/app.ts
Comment (Archit404Error) [inline src/app.ts:118]: You should use the Organization Controller here instead, per MVC design pattern

File: src/app.ts
Comment (Archit404Error) [inline src/app.ts:111]: You can store the required fields as strings in an array and use `.every()` on the array to check if the value is contained in the object's keys. An example of this is [here](https://stackoverflow.com/questions/55871220/how-to-check-if-a-json-doc-has-all-required-keys-in-node-js).

I honestly think what you have is fine, though, it's probably more readable than the other approach

File: src/app.ts
Comment (Archit404Error) [inline src/app.ts:115]: Have you checked that the upload service can handle a massive Base64 string sent via a network request (i.e., have you tried uploading a really large file and seeing if that works)

File: src/app.ts
Comment (Archit404Error) [inline src/app.ts:84]: Yeah `/flyers/` is fine, I think this location is fine for now since I don't foresee us adding more flyer functionality. If we do, we'll be able to justify having a separate folder for flyer code and can refactor this then

**Review by Archit404Error — APPROVED** [review]
Great work on this Zach! Just a few small comments, otherwise feel free to remove the comments in the code and you should be good to go! Also in the future I'd suggest a more descriptive PR title (you can still change it before you merge) and also if you do have comments about certain chunks of code rather than putting the comments directly in the code you can just make comments on GitHub about the relevant sections after making the PR

File: src/app.ts
Comment (zachseidner1) [reply src/app.ts:115]: I checked it just now, and it crashes the backend as utils throws an unchecked exception. Not really sure what to do considering the upload service is sort of a black box since we're using AppDev's upload service and not s3 or something. Do we need to rewrite AppDev's upload service to use form-data too? I feel like there's not much I can change on Volume's backend to fix this other than wrapping in a try catch.

Comment (Archit404Error) [reply src/app.ts:115]: Yeah I spoke about this with Vin but we would have to add an endpoint to the upload service that can handle a form request, and then you can just forward req.file as part of a form request to the endpoint in the upload service

File: src/app.ts
Comment (zachseidner1) [reply src/app.ts:118]: my bad, didn't realize this existed!

File: src/app.ts
Comment (zachseidner1) [reply src/app.ts:111]: right, but I can't do that right now since the upload service can't handle a form right now, so is this ok for right now?

Comment (Archit404Error) [reply src/app.ts:111]: The issue we're trying to solve is that certain orgs can't upload files because the files are too large right? So if we can't deal with large files then we would still have this issue even if we merge this code right? I think if we want to resolve the file size issue once and for all we would have to make that new endpoint in the upload service (which should hopefully be just a couple lines of code because I think `boto3` which is used in the upload service has support for this).

Comment (zachseidner1) [reply src/app.ts:111]: I agree for sure, I just wasn't sure if I had access to change the upload service. I found the GitHub repository and started working on a fix. I will also update this PR again to use form data in the post request to the upload service once I am done.

**Review by vinnie4k — COMMENTED** [review]
Great work! We should definitely take a look at the upload service before merging this to ensure that large B64 strings are handled. We should add an endpoint to our upload service that takes in a form request.

**Review by vinnie4k — APPROVED** [review]

**Pull request merged**
